### PR TITLE
Scripts: Copy PHP files from `src` into `build`

### PR DESCRIPTION
**Pull request by ryanwelcher**
## Description
When building dynamic blocks, there are many cases where splitting out the markup into an external PHP file is very helpful. The current build process copies the block.json file into the `build` directory but any `.php` files are not. This PR adds them to the copying process. 

## Testing Instructions
I created a block using `npx @wordpress/create-block@ my-first-dynamic-block --template=@ryanwelcher/dynamic-block-template` ( this contains an template.php file in the src directory )
Then ran `node {FULLPATH}/gutenberg/packages/scripts/bin/wp-scripts.js -- build`

## Types of changes
Adds any .PHP files inside of `src` to it's corresponding location in `build`

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [x] I've updated related schemas if appropriate.

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
Looks great. I added a changelog entry since people will suddenly start seeing PHP files in the build folder which might be surprising for them.

**Comment by ryanwelcher** [comment]
> Looks great. I added a changelog entry since people will suddenly start seeing PHP files in the build folder which might be surprising for them.

Thanks, @gziolo! I ALWAYS forget to add the changelog.

**Comment by herndlm** [comment]
Hi, is there a way to disable this? I guess I need to override the webpack config?

**Comment by gziolo** [comment]
> Hi, is there a way to disable this? I guess I need to override the webpack config?

Yes, you would need to override the default config. Something like:

https://github.com/WordPress/gutenberg/tree/trunk/packages/scripts#extending-the-webpack-config

Can you explain the issue you have? It looks like the change introduced needs some tweaks so we are seeking more detailed feedback to provide better defaults.

**Comment by herndlm** [comment]
Sure, I'm happy to do so. We're using `wordpress-scripts` in some ouf WordPress PHP plugins where we also have JS parts. Maybe this is already a bit controversial. We have already some customization for that, but basically we have
- some generic PHP classes in `src/`
- a `plugin-name.php` in root, that adds e.g. a couple of shortcodes or bootstraps stuff that uses the classes from `src/`
- a TypeScript app in `assets/src`, for that one we use wordpress-scripts for testing, linting and building/transpiling it to public from where we load / enqueue it in plugin-name.php with plain WP code

So basically, our use-case is using `wordpress-scripts` in a WordPress environment that is not Gutenberg :) And in that case I don't want the generic PHP classes end up in public.

**Comment by delawski** [comment]
> We have already some customization for that, but basically we have
> 
> * some generic PHP classes in `src/`
> * a `plugin-name.php` in root, that adds e.g. a couple of shortcodes or bootstraps stuff that uses the classes from `src/`
> * a TypeScript app in `assets/src`, for that one we use wordpress-scripts for testing, linting and building/transpiling it to public from where we load / enqueue it in plugin-name.php with plain WP code

We're using a similar directory structure in [AMP WP plugin](https://github.com/ampproject/amp-wp), i.e. the `src` directory stores PHP files while JS files are kept in `assets/src` and we're having analogical issues: ampproject/amp-wp#6947.

Right now the default source directory name of `src` is hardcoded in multiple places in the package (e.g. in the [`getWebpackEntryPoints()`](https://github.com/WordPress/gutenberg/blob/trunk/packages/scripts/utils/config.js#L180) or in the [`webpack.config.js`](https://github.com/WordPress/gutenberg/blob/trunk/packages/scripts/utils/config.js#L187)). I guess it'd be great to make it configurable in some way.

**Comment by ryanwelcher** [comment]
Follow up PR to move to an opt-in approach - #39171

**Comment by gziolo** [comment]
> Follow up PR to move to an opt-in approach - #39171

Landed, I will publish shortly to npm to remove friction as discussed.
